homeassistant/components/lastfm/sensor.py

```diff
@@ -60,11 +89,24 @@ class LastFmSensor(SensorEntity):
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user: User, lastfm_api: LastFMNetwork) -> None:
+    def __init__(self, user: User, entry_id: str) -> None:
         """Initialize the sensor."""
-        self._attr_unique_id = hashlib.sha256(user.name.encode("utf-8")).hexdigest()
-        self._attr_name = user.name
         self._user = user
+        self._attr_unique_id = hashlib.sha256(user.name.encode("utf-8")).hexdigest()
+        self._attr_name = f"lastfm_{user.name}"
+        try:
+            user.get_playcount()
+        except WSError as exc:
+            self._attr_available = False
+            LOGGER.error("Failed to load LastFM user `%s`: %r", user.name, exc)
+            return
+        self._attr_device_info = DeviceInfo(
+            configuration_url="https://www.last.fm",
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer=DEFAULT_NAME,
+            name=DEFAULT_NAME,
```

Comment:
Should probably append `user.name` as otherwise you end up with several devices which are all called `LastFM`